source/MRMesh/MRMeshOrPoints.h

```diff
@@ -117,6 +130,25 @@ MRMESH_API void projectOnAll(
     const ProjectOnAllCallback & callback, ///< each found closest point within given distance will be returned via this callback
     ObjId skipObjId = {} ); ///< projection on given object will be skipped
 
+/// Projects a point onto an object, in world space. Returns `.valid() == false` if this object type isn't projectable onto.
+[[nodiscard]] inline MeshOrPoints::ProjectionResult projectWorldPointOntoObject( const Vector3f& p, const Object& obj )
+{
+    auto mop = getMeshOrPointsXf( &obj );
+    if ( !mop )
+        return {};
+    return mop->projector()( p );
+}
+
+/// Recursively visits the objects and projects the point on each one. Returns the closest projection.
+/// If `projectPred` is specified and false, will not project onto this object.
+/// If `recursePred` is specified and false, will not visit the children of this object.
+[[nodiscard]] MRMESH_API MeshOrPoints::ProjectionResult projectWorldPointOntoObjectsRecursive(
+    const Vector3f& p,
+    const Object& root = SceneRoot::get(),
```

Comment:
Please avoid `SceneRoot` in this header.
Better make `const Object* root = nullptr`, which is interpreted as `SceneRoot` inside .cpp